### ISSUE TITLE
Allow compressed result file upload

### ISSUE THREAD

**Issue by janisliepins** [fescobar/allure-docker-service#149]
It would be great to be able to POST compressed test result files (.zip, .tar formats)
Need for this arises when i am trying to send several thousand result files to /send-results endpoint.

**Comment by fescobar** [comment]
@janisliepins what is the problem sending multiple files?

**Comment by janisliepins** [comment]
> @janisliepins what is the problem sending multiple files?

Well in my case i have several thousand of those result files (in one case around 11 000). So i had to modify some flask options like max_request_body_size for app to be able to handle all those files ... + had issues with allure service host VM where regularly inode limit was reached for the filesystem because of the enormous file count.

This could be all resolved with result archiving - e.g. like for Jenkins Allure plugin - where Jenkins creates ZIP archive for Allure report.

Also question and possible solution - maybe i can send Junit report.xml file to the Allure service? Is it possible? have not tried it yet ....

**Comment by kormachevt** [comment]
Would be a superb feature. I stumbled upon 'Argument list is too long' error from curl during an attempt to send over 4000 files. 
Currently my workaround is splitting array, but archiving the report would solve this problem entirely.

**Comment by arekmadej** [comment]
I've also started to see some issues around sending large amounts of files leading to reliability issues to our nightly automation runs. I'm happy to implement the functionality to allow sending compressed files to the API endpoint and handing it accordingly as long as @fescobar is ok with this

**Comment by fescobar** [comment]
@arekmadej Go ahead, please. Fork the project and when you finished your implementation, please create a pull request again my beta branch. I will test it in there.
Also, it would be good if you can add an example here::
https://github.com/fescobar/allure-docker-service#send-results-through-api

Thank you in advance

**Comment by fescobar** [comment]
@arekmadej the flow will be the next:
1. POST https://github.com/fescobar/allure-docker-service/blob/master/allure-docker-api/app.py#L851
2. Validate extension of file zip received
3. Put the zip inside the container in path /tmp and extract it in there with a directory with a UID name.
4. Move the files extracted to the allure-results directory

You need to move first the zip to the temporal directory, otherwise, the automatic check results will be executed and will generate another report. For that reason, don't move directly the zip to the allure-results directory.

Any questions let me know.

**Comment by antoniordz96** [comment]
+1. Is there any recommendation or workaround currently?

**Comment by boazbarda** [comment]
My issue was a javaheap when reached a lot of files. I assume it was thier size and not the amount.
I workaround it by sending the results in batches and generate at the end.

**Comment by ihormanzii** [comment]
I have solved this by uploading files in batches

```
#!/bin/bash

batch_size=50
total=${#files_to_send[@]}
batches=$(( (total + batch_size - 1) / batch_size ))

echo "Batch size: $batch_size" >&2
echo "Total files to upload: $total" >&2

for ((i = 0; i < total; i += batch_size)); do
  batch=("${files_to_send[@]:i:batch_size}")
  files=()
  for file in "${batch[@]}"; do
    files+=(-F "files[]=@${file}")
  done

  echo "Uploading batch $((i / batch_size + 1)) of $batches (${#batch[@]} files)..." >&2

  upload_status=$(curl --retry 5 -sS -w "%{http_code}" \
    -o /tmp/upload_body.txt \
    -X POST "$ALLURE_SERVER/allure-docker-service/send-results?project_id=$PROJECT_NAME&force_project_creation=true" \
    -H 'Content-Type: multipart/form-data' \
    "${files[@]}" -k)

  if [ "$upload_status" -ne 200 ]; then
    echo "Upload failed on batch $((i / batch_size + 1)) with status $upload_status:" >&2
    cat /tmp/upload_body.txt >&2
    exit 1
  fi
done
```

**Comment by evsasha** [comment]
I am concerned that such a significant improvement has not yet been merged. From my perspective, there are no objective reasons to delay this implementation further.

* **User Experience**: Shifting the complexity of report delivery to the user by requiring them to write extensive scripts for splitting payloads into parts seems counterintuitive.

* **Flexibility vs. Requirement:** It is important not to confuse an alternative option (sending archives) with a workaround. Implementing archive support does not force users to change their current workflow; those who prefer the existing method can continue using it as usual.

* **Efficiency and Scaling:** Since report data is text-based, archiving provides significant data compression. Reducing the payload size leads to faster transmission, which directly addresses one of the key challenges of scaling the service.

Bottom line: I am currently using two solutions in parallel:

* This project, with a custom patch for archive support.

* The [allure-server](https://github.com/kochetkov-ma/allure-server) project, which has native support for sending data via archives.

I believe merging this feature would greatly benefit the community and the project's usability.

**Comment by fescobar** [comment]
> I am concerned that such a significant improvement has not yet been merged. From my perspective, there are no objective reasons to delay this implementation further.
> 
> * **User Experience**: Shifting the complexity of report delivery to the user by requiring them to write extensive scripts for splitting payloads into parts seems counterintuitive.
> * **Flexibility vs. Requirement:** It is important not to confuse an alternative option (sending archives) with a workaround. Implementing archive support does not force users to change their current workflow; those who prefer the existing method can continue using it as usual.
> * **Efficiency and Scaling:** Since report data is text-based, archiving provides significant data compression. Reducing the payload size leads to faster transmission, which directly addresses one of the key challenges of scaling the service.
> 
> Bottom line: I am currently using two solutions in parallel:
> 
> * This project, with a custom patch for archive support.
> * The [allure-server](https://github.com/kochetkov-ma/allure-server) project, which has native support for sending data via archives.
> 
> I believe merging this feature would greatly benefit the community and the project's usability.

I'm working in a big big improvement that will take time. There is no any existing PR that has the right solution.
